cni/pkg/log/uds.go

```diff
@@ -108,29 +109,68 @@ func (l *UDSLogger) processLog(body []byte) {
 	}
 	messages := make([]cniLog, 0, len(cniLogs))
 	for _, l := range cniLogs {
-		var msg cniLog
-		if err := json.Unmarshal([]byte(l), &msg); err != nil {
-			log.Debugf("Failed to unmarshal CNI plugin log entry: %v", err)
+		msg, ok := parseCniLog(l)
+		if !ok {
 			continue
 		}
-		msg.Msg = strings.TrimSpace(msg.Msg)
 		messages = append(messages, msg)
 	}
 	// Lock log message printing to prevent log messages from different CNI
 	// processes interleave.
 	l.mu.Lock()
 	defer l.mu.Unlock()
 	for _, m := range messages {
+		logger := pluginLog
+		// For any k/v pairs, add them back
+		for k, v := range m.Arbitrary {
+			logger = logger.WithLabels(k, v)
+		}
 		// There is no fatal log from CNI plugin
 		switch m.Level {
 		case "debug":
-			pluginLog.LogWithTime(istiolog.DebugLevel, m.Msg, m.Time)
+			logger.LogWithTime(istiolog.DebugLevel, m.Msg, m.Time)
 		case "info":
-			pluginLog.LogWithTime(istiolog.InfoLevel, m.Msg, m.Time)
+			logger.LogWithTime(istiolog.InfoLevel, m.Msg, m.Time)
 		case "warn":
-			pluginLog.LogWithTime(istiolog.WarnLevel, m.Msg, m.Time)
+			logger.LogWithTime(istiolog.WarnLevel, m.Msg, m.Time)
 		case "error":
-			pluginLog.LogWithTime(istiolog.ErrorLevel, m.Msg, m.Time)
+			logger.LogWithTime(istiolog.ErrorLevel, m.Msg, m.Time)
+		}
+	}
+}
+
+// parseCniLog is tricky because we have known and arbitrary fields in the log, and Go doesn't make that very easy
+func parseCniLog(l string) (cniLog, bool) {
```

Comment:
I actually wonder if this should eventually end up in our bespoke logging library - we are at the end of the day serializing and deserializing our own format, and having a generic way to unmarshal our own log format from a string is probably useful.